katas/content/multi_qubit_gates/two_qubit_gate_2/solution.md

```diff
@@ -0,0 +1,49 @@
+Firstly we notice that we are dealing with an unentangled pair of qubits. In vector form this is:
+In vector form the transformation we need is
+
+In vector form the transformation we need is 
+$$
+\frac{1}{2}\begin{bmatrix}1\\\ 1\\\ 1\\\ 1\\\ \end{bmatrix} 
+\rightarrow 
+\frac{1}{2}\begin{bmatrix}1\\\ 1\\\ 1\\\ -1\\\ \end{bmatrix}
+$$
+
+All that needs to happen to change the input into the goal is that the $|11\rangle$ basis state needs to have its sign flipped.
+
+We remember that the Pauli Z gate flips signs in the single qubit case, so we need to investigate if there is a 2-qubit version of this gate that we can use here. We can also recall task "Phase i" which dealt with phase shifts and, remembering that $e^{i\cdot\pi} = -1$, we can think of the transformation we're looking for as a phase shift.
+It can be useful to investigate a general case and then use it to perform a specific state change, so let's look for a 2-qubit variant of the phase shift.
+Similarly to task "Two qubit gate 1", the phase shift only occurs on one of the basis states, so this suggests it might be a conditional shift. If we could have our phase shift applied to `qs[1]` conditional on `qs[0]` being in the state $|1\rangle$, then we would have a description of our gate. If we now look through a list of gates in the Single-qubit gates tutorial, we'll find the R1 phase shift gate with angle parameter $\theta$ (radians), defined as
+
+$$
+R1(\alpha)= 
+ \begin{bmatrix}1 & 0\\\ 0 & \color{red}{e^{i\alpha}} \end{bmatrix}
+$$
+
+The controlled variant of this gate will look like this:
+
+$$
+CR1(\alpha) = 
+ \begin{bmatrix}1 & 0 & 0 & 0\\\ 0 & 1 & 0 & 0\\\ 0 & 0 & 1 & 0\\\ 0 & 0 & 0 &\color{red}{e^{i\alpha}} \end{bmatrix}
+$$
+
+This gate is almost Pauli I, the identity gate, with the different in just the last column, showing what will happen to the $|11\rangle$ basis state. Applying it to our input state for $\alpha = \pi$, we'll get:
+
+$$
+\frac{1}{2}\begin{bmatrix}1 & 0 & 0 & 0\\\ 0 & 1 & 0 & 0\\\ 0 & 0 & 1 & 0\\\ 0 & 0 & 0 &\color{red}{e^{i\alpha}} \end{bmatrix}
+\begin{bmatrix}1\\\ 1\\\ 1\\\ 1\\\ \end{bmatrix}=
+\frac{1}{2}\begin{bmatrix}1\\\ 1\\\ 1\\\ 1\cdot\color{red}{e^{i\alpha}}\\ \end{bmatrix}=
+\frac{1}{2} \big( |00\rangle + |01\rangle + |10\rangle {\color{red}-} |11\rangle \big)
+$$
+
+The last thing we notice if we look through the [list of operations in the Microsoft.Quantum.Canon namespace](https://learn.microsoft.com/en-us/qsharp/api/qsharp-lang/microsoft.quantum.canon/cz) is the CZ (Controlled Z) gate, a special case of CR1 that implements exactly this gate.
```

Comment:
I'm not really sure why the explanation in the workbook is so convoluted when it leads to just using CZ... It's possible that we had in mind an extra task that would introduce a complex relative phase, but it's rather an overkill in this case with the CZ lesson added. Let's shorten it

```suggestion
We remember that the Pauli $Z$ gate flips signs in the single qubit case, and that $CZ$ is the 2-qubit version of this gate. And indeed, the effect of the $CZ$ gate is exactly the transformation we're looking for here.
```